src/mats_l1_processing/instrument.py

```diff
@@ -332,29 +351,13 @@ def __init__(self, channel, calibration_file, start_datetime: datetime = datetim
 
         # single event correction
         filename = calibration_data['hot_pixels']['single_events']
-        disk_conn = sqlite.connect(filename)
-        query = f'''
-            SELECT * FROM SingleEvents
-            WHERE datetime BETWEEN '{start_datetime}' AND '{end_datetime}'
-            AND channel = '{self.channel}'
-        '''
-        self.single_event = pd.read_sql_query(query, disk_conn)
-        self.single_event['datetime'] = pd.to_datetime(self.single_event['datetime'])
-        disk_conn.close()
+        self.single_event = get_sqlite_data(filename, "SingleEvents")
 
         # hot pixel correction
         filename = calibration_data['hot_pixels']['hot_pixels']
-        disk_conn = sqlite.connect(filename)
-        query = f'''
-            SELECT * FROM hotpixelmaps
-            WHERE datetime BETWEEN '{start_datetime}' AND '{end_datetime}'
-            AND channel = '{self.channel}'
-        '''
-        self.hot_pixels = pd.read_sql_query(query, disk_conn)
-        self.hot_pixels['datetime'] = pd.to_datetime(self.hot_pixels['datetime'])
+        self.hot_pixels = get_sqlite_data(filename, "hotpixelmaps")
         # Convert the HPM to a NumPy array
-        self.hot_pixels['HPM'] = self.hot_pixels['HPM'].apply(deserialize_blob)
-        disk_conn.close()
+        self.hot_pixels['HPM'] = self.hot_pixels['HPM'].apply(pickle.loads)
```

Comment:
pickle.loads did not work for me (when I tried), but perhaps I did something wrong..

Reply:
Good to know!

Reply:
Seems to work in the production environment, but would probably be good to test it locally as well.